pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -1048,6 +1084,43 @@ func (m *ManagerImpl) GetDevices(podUID, containerName string) ResourceDeviceIns
 	return m.podDevices.getContainerDevices(podUID, containerName)
 }
 
+func (m *ManagerImpl) UpdateAllocatedResourcesStatus(pod *v1.Pod, status *v1.PodStatus) {
+	for i, containerStatus := range status.ContainerStatuses {
+		devices := m.podDevices.getContainerDevices(string(pod.UID), containerStatus.Name)
+
+		for resourceName, deviceInstances := range devices {
+			for id, d := range deviceInstances {
+				d.Health = m.allDevices[resourceName][id].Health
+
+				deviceInstances[id] = d
```

Comment:
I can't say the following is better, but this `for` loop above should be equivalent to this one, right?
```
for id := range deviceInstances {
   deviceInstances[id].Health = m.allDevices[resourceName][id].Health
}
```

Reply:
(asking to doublecheck my understanding, mostly)

Reply:
Yes, but suggested gives: `cannot assign to struct field deviceInstances[id].Health in mapcompiler[UnaddressableFieldAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#UnaddressableFieldAssign)`

Reply:
ok ok, we can keep as you have now.